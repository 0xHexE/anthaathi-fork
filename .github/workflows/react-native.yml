name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
