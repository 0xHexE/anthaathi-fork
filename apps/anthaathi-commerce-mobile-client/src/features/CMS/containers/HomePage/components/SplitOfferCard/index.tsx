import {View, Image} from 'react-native';
import React from 'react';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';
import {Button, Text} from 'react-native-paper';
import {HomePageComponentType} from '../../../../types/common';

export interface SplitCardOfferProps {
  title: string;
  subtitle: string;
  description?: string;
  image: string;
  buttonTitle: string;
  onPress?: () => void;
}

const SplitOfferCard = (props: SplitCardOfferProps) => {
  const itemWidth = useResponsiveValue(['90%', '60%', '50%', '50%']);
  const itemHeight = useResponsiveValue([240, 280, 320, 320]);
  const itemTwoWidth = useResponsiveValue(['80%', '40%', '50%', '50%']);

  return (
    <View
      style={{
        flexDirection: itemWidth === '90%' ? 'column' : 'row-reverse',
        alignItems: 'center',
        marginVertical: 10,
      }}
      testID="splitCardOffer">
      <Image
        testID="offerImage"
        source={{
          uri: props.image,
        }}
        style={{
          width: itemWidth,
          height: itemHeight,
        }}
      />
      <View style={{width: itemTwoWidth, alignItems: 'center'}}>
        <Text
          style={{
            color: '#000',
            fontSize: 22,
            fontWeight: 'bold',
            marginVertical: 5,
            paddingTop: 10,
          }}
          variant="titleSmall">
          {props.title}
        </Text>

        <Text
          style={{
            textAlign: 'center',
            color: '#000',
            fontSize: 16,
            fontWeight: '100',
            marginVertical: 10,
          }}
          variant="titleSmall">
          {props.subtitle}
        </Text>
        <Button
          testID="buttonPress"
          mode="contained"
          style={{
            backgroundColor: '#000000',
            borderRadius: 1,
            marginVertical: 5,
          }}
          onPress={props.onPress}>
          {props.buttonTitle}
        </Button>
      </View>
    </View>
  );
};

export default SplitOfferCard;

export const SplitOfferCardCMSInput = {
  _component: HomePageComponentType.SplitOfferCard,
  component: SplitOfferCard,
};
