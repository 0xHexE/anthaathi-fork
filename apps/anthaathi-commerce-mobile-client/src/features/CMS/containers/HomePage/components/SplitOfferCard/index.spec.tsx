import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import SplitOfferCard from './index';

describe('SplitOfferCard', () => {
  const image =
    'https://cdn.shopify.com/s/files/1/0648/1303/9842/files/Newsletter_bg_300x.png?v=1653648705';
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SplitOfferCard
            title="Get Exclusive Offers"
            subtitle="Get exclusive offers & more by signing up for our promotional email"
            image={image}
            buttonTitle="View Offers"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('splitCardOffer')).toBeTruthy();
  });

  it('should call when we call tap onPress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SplitOfferCard
            title="Get Exclusive Offers"
            subtitle="Get exclusive offers & more by signing up for our promotional email"
            image={image}
            buttonTitle="View Offers"
            onPress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('buttonPress')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have SplitOfferCard title, subtitle and button title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SplitOfferCard
            title="Get Exclusive Offers"
            subtitle="Get exclusive offers & more by signing up for our promotional email"
            image={image}
            buttonTitle="View Offers"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Get Exclusive Offers')).toBeTruthy();
    expect(
      temp.queryByText(
        'Get exclusive offers & more by signing up for our promotional email',
      ),
    ).toBeTruthy();
    expect(temp.queryByText('View Offers')).toBeTruthy();
  });

  it('should have SplitOfferCard Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <SplitOfferCard
            title="Get Exclusive Offers"
            subtitle="Get exclusive offers & more by signing up for our promotional email"
            image={image}
            buttonTitle="View Offers"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('offerImage')!.props.source).toMatchObject({
      uri: image,
    });
  });
});
