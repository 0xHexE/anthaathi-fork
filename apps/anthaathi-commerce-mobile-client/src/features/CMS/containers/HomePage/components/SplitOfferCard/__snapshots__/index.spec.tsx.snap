// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitOfferCard should render the item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row-reverse",
      "marginVertical": 10,
    }
  }
  testID="splitCardOffer"
>
  <Image
    source={
      Object {
        "uri": "https://cdn.shopify.com/s/files/1/0648/1303/9842/files/Newsletter_bg_300x.png?v=1653648705",
      }
    }
    style={
      Object {
        "height": 280,
        "width": "60%",
      }
    }
    testID="offerImage"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "40%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#000",
            "fontSize": 22,
            "fontWeight": "bold",
            "marginVertical": 5,
            "paddingTop": 10,
          },
        ]
      }
    >
      Get Exclusive Offers
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "color": "#000",
            "fontSize": 16,
            "fontWeight": "100",
            "marginVertical": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Get exclusive offers & more by signing up for our promotional email
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#000000",
              "borderColor": "transparent",
              "borderRadius": 1,
              "borderStyle": "solid",
              "borderWidth": 0,
              "marginVertical": 5,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 1,
                },
              ]
            }
            testID="buttonPress"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ]
                }
              >
                View Offers
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
